packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4369,67 +4369,343 @@ class PositionedDirectional extends StatelessWidget {
+/// in the main axis.
+///
+/// Any time a widget is given "tight constraints" by its parent,
+/// that widget will be forced to use all of the space offered to it.
+///
+/// [FlexFit.loose] is the default fit for the [Flexible]. It will not cause
+/// its child to stretch in the main axis, and will allow the child to be the
+/// size the child's parameters state it should be.
+///
+/// [MainAxisSize] : An enum that that determines if the [Flex] will
+/// be forced to be as large as its parent will allow.
+///
+/// [MainAxisSize.max] will force the [Flex] to be as large as the parent
+/// will allow in the main axis. _This does not force the children to be
+/// larger._
+///
+/// [MainAxisSize.min] : Allow the [Flex] to be whatever size it needs
+/// to be in order to accommodate its children, without forcing the
+/// [Flex] to be any larger than the total size of all its children.
+///
+/// [MainAxisAlignment] : How to align, center, or apply spacing
+/// between the children in the direction in the main axis.
+///
+/// [CrossAxisAlignment] : How to align, or stretch, the children in
+/// the direction of the cross axis.
+///
+/// Constraints: These are the minimum and maximum sizes between
+/// which a widget must be. There are constraints both vertically and
+/// horizontally. If constraints are _tight_, this means the min and max
+/// limits are the same, forcing the widget to be that specific size. If
+/// constraints are loose, the widget can be any size within the specified
+/// min and max, for both width and height.
+///
+/// There is another case in which there are no constraints given,
+/// and this can be the case with the children of a [Flex].
+/// This is also the source of many errors seen while using [Row]s
+/// and [Column]s. Error messages referring to this will often use
+/// the term _unbounded_ to describe the situation in which no
+/// constraints (min or max size limits) have been passed down from
+/// the parent.
+///
+/// Now that all of those terms have been explained, the multi-child
+/// layout algorithm for a [Flex] proceeds in six steps:
+///
+/// 1. Layout each child that is not a [Flexible] or an [Expanded],
+///    but do so without setting any minimum or maximum size in the
+///    main axis. This is called laying the child out "without applying
+///    any constraints in the main axis".
+///
+///    Apply any constraints in the _cross axis_ direction that may have
+///    been passed down by the parent of the [Flex], without forcing the
+///    child to be any larger than it's stated size in that direction. This is
+///    called using _"loose constraints"_ (in this case, in the cross axis).
+///
+///    However, if the [CrossAxisAlignment] of the [Flex] has been
+///    set to [CrossAxisAlignment.stretch] then force all the children of the
+///    [Flex] to be as large as possible in the cross axis direction. This is
+///    called using _"tight constraints"_.
```

Comment:
I mean, yes, it uses tight constraints on the children to make them all be as large as they can be, but that's not the definition of using "tight constraints". Using tight constraints just means that the max and the min values are identical, so there's not confusion about what size it can be: it can only be one size.